src/atlas_frontend/src/canisters/atlasSpace/api.ts

```diff
@@ -195,6 +199,28 @@ export const rejectSubtaskSubmission = async ({
   });
 };
 
+export const getRejectionInfo = (
+  submissionData: SubmissionData,
+  submissionState: "Rejected" | "WaitingForReview" | "Accepted"
+) => {
+  const rejectionReasonArray = submissionData.rejection_reason;
+
+  const ShowRejectionReason =
+    submissionState === "Rejected" &&
+    Array.isArray(rejectionReasonArray) &&
+    rejectionReasonArray.length > 0 &&
+    typeof rejectionReasonArray[0] === "string" &&
+    rejectionReasonArray[0].length > 0 &&
+    rejectionReasonArray[0].trim().length > 0;
```

Comment:
This is very **rough** and should be **handled** much better. For example:

```suggestion
  const rejectionReason = submissionData.rejection_reason.pop();
  const ShowRejectionReason =
    submissionState === "Rejected" &&
    rejectionReason &&
    rejectionReason.length > 0
```

But **everything** regarding **handling** optional **things** (`submissionData.rejection_reason`) should be handled in an `api.ts` file.
When generating a communication interface from Candid, Rust `Option<T>` is translated to TypeScript `[T]`.
This means that we should **treat** this not as a list but as a **tuple** that may have only one item in it or be empty.

Reply:
And use camelCase not a PascalCase for variable naming